contracts/StablecoinConverter.sol

```diff
@@ -209,20 +215,46 @@ contract StablecoinConverter is EpochTokenLocker {
                 executedSellAmount
             );
         }
-        checkAndOverrideObjectiveValue(uint(tokenConservation[0]));
+        int objectiveValue = utility - disregardedUtility;
+        checkAndOverrideObjectiveValue(objectiveValue);
         grantRewardToSolutionSubmitter(uint(tokenConservation[0]) / 2);
         checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
 
-    function getCurrentObjectiveValue() public view returns(uint) {
+    function getCurrentObjectiveValue() public view returns(int) {
         if (previousSolution.batchId == getCurrentBatchId() - 1) {
             return previousSolution.objectiveValue;
         } else {
-            return 0;
+            return MIN_INT;
         }
     }
 
+    function evaluateUtility(uint128 execBuy, uint128 execSell, Order memory order) internal view returns(int) {
+        // Utility = ((execBuyAmt * order.sellAmt - execSellAmt * order.buyAmt) * price.buyToken) / order.sellAmt
+        (uint128 buyAmount, uint128 sellAmount) = getBuySellAmounts(order);
+        return ((execBuy * sellAmount - execSell * buyAmount) * currentPrices[order.buyToken]) / sellAmount;
+    }
+
+    function evaluateDisregardedUtility(uint128 execSell, Order memory order) internal view returns(int) {
+        // |disregardedUtility| = maxUtility - actualUtility
+        // where maxUtility is the utility achieved when execSellAmount == order.sellAmount. That is,
+        // maxUtility = ((execBuyAmt * order.sellAmt - order.sellAmt * order.buyAmt) * price.buyToken) / order.sellAmt
+        //            = (execBuyAmt - order.buyAmount) * price.buyToken <---- This reduction not used in final formula.
+        // So, after simplification
```

Comment:
Does it make sense to show the simplification here? It's not super obvious why execBuyAmt is no input to this function, but I guess the reader can also do the math himself?

Reply:
This was just included for my reviewers. Will disappear before merge!